packages/redux/src/raffles/reducer.ts

```diff
@@ -0,0 +1,173 @@
+import * as actionTypesImports from './actionTypes';
+import { AnyAction, combineReducers, Reducer } from 'redux';
+import createReducer, {
+  createReducerWithResult,
+} from '../helpers/reducerFactory';
+import type * as T from './types';
+import type {
+  CreateRaffleParticipationAction,
+  FetchRaffleAction,
+  FetchRaffleEstimationAction,
+  FetchRaffleParticipationAction,
+  RafflesState,
+  UpdateRaffleParticipationAction,
+} from './types';
+import type { StoreState } from '../types';
+
+const {
+  RAFFLES_ACTIONS,
+  RAFFLE_ACTIONS,
+  PARTICIPATION_ACTIONS,
+  ESTIMATION_ACTIONS,
+  PARTICIPATION_CREATIONS_ACTIONS,
+  ...actionTypes
+} = actionTypesImports;
+
+export const INITIAL_STATE: T.RafflesState = {
+  allRaffles: {
+    result: null,
+    isLoading: false,
+    error: null,
+  },
+  raffles: {},
+  estimations: {},
+  participations: {},
+  participationCreations: {},
```

Comment:
What is the difference between `participations` and `participationsCreations`? I find this a bit confusing and it is not explained anywhere.

Reply:
The `participations` piece of state is an object that stores `{ isLoading: boolean, error: BlackoutError | null }` like objects indexed by `participationId`s so the user can keep track of the state of individual fetch participation requests.
The `participationsCreations` piece of state is an object that stores `{ isLoading: boolean, error: BlackoutError | null }` like objects indexed by `raffleId`s so that the user can keep track of the of individual create participation requests. Because at creating time of participations, there is no `participationId` to index such metadata, and because each user should only have one participation per raffle, `rafflesId` is being used to index this piece of state.

Reply:
Got it! Thanks